src/clib/rate_functions.c

```diff
@@ -0,0 +1,1307 @@
+/**********************************************************************
+ * ---- File containing definitions of all rates ----
+ *
+ * Written by: Ewan Jones
+ * Date of Creation: 26/4/2021
+ * 
+ * Purpose:
+ *  Contains definitions for all rates. These will be called within 
+ *  calc_rates_g_c.
+ * 
+ * Set "units" parameter = 1 for cgs unit outputs.
+**********************************************************************/
+
+//* Macro definitions for constants.
+//Kelvin to eV conversion factor
+#ifndef tevk
+#define tevk 1.1605e4
+#endif
+//Comparison value
+#ifndef dhuge
+#define dhuge 1.0e30
+#endif
+
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "grackle_macros.h"
+#include "grackle_types.h"
+#include "grackle_chemistry_data.h"
+#include "cie_thin_cooling_rate_tables.h"
+#include "phys_constants.h"
+
+
+// Calculation of k1.
```

Comment:
This is an heroic translation and so I'm hesitant to ask for more work, but I'm wondering if it wouldn't improve readability a great deal to indicate the actual reaction (e.g. for k7, write: HI + e -> H2I + gamma) associated with each rate?

Reply:
The same thought crossed my mind at one point, but I must have forgotten about it. I think this would be a nice idea too, I will add this to my todo's.